Source/AdUnits/Native/NativeAsset.swift

```diff
@@ -0,0 +1,143 @@
+/*   Copyright 2018-2019 Prebid.org, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import UIKit
+
+public class NativeAsset: NSObject {
+    
+    var id: NSInteger!
+    public var required: Bool = false
+    public var assetObject:AnyObject?
+    
+    public required init?(isRequired: Bool, nativeObject:AnyObject) {
+        super.init()
+        required = isRequired
+        var containsAssetType = true
+        
```

Comment:
init method should not return nil? I'm not quite familiar with swift syntax but we can use factory create pattern here?

Reply:
modified this initializer